### PR TITLE
[cargo-license] Add cargo-license

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

**Comment by giordano** [comment]
> Fix nproc syntax

???

**Comment by jeremiahpslewis** [comment]
> > Fix nproc syntax
> 
> ???

Github auto-title. Fixing.

File: C/cargo_license/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+
+cd $WORKSPACE/srcdir/cargo-license
+
+cargo build --release --all
+
+mkdir ${bindir}
+cp target/${rust_target}/release/cargo-license${exeext} ${bindir}/
+
+cargo-license --avoid-build-deps --avoid-dev-deps --json > CARGO_LICENSES.json
```
Comment (giordano) [inline C/cargo_license/build_tarballs.jl:24]: Am I missing something or this is going to fail again?

Comment (jeremiahpslewis) [reply C/cargo_license/build_tarballs.jl:24]: This is the 'bootstrap' call of cargo-license on the cargo-license library I build with this script

Comment (jeremiahpslewis) [reply C/cargo_license/build_tarballs.jl]: But it does fail for other reasons. ;)

Comment (jeremiahpslewis) [reply C/cargo_license/build_tarballs.jl]: five minutes later....You aren't missing anything.

**Comment by jeremiahpslewis** [comment]
@giordano Once this passes and merges, can I use a `BuildDependency("cargo_license)` in this project? Does that count as a legitimate bootstrap or does some ugly form of infinite recursion result...

**Comment by giordano** [comment]
As I [said already](https://github.com/JuliaPackaging/Yggdrasil/pull/4088#discussion_r771856742), that must be a [`HostBuildDependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.HostBuildDependency). See also https://docs.binarybuilder.org/stable/build_tips/#Dependencies-for-the-target-system-vs-host-system

**Comment by jeremiahpslewis** [comment]
👍

File: C/cargo_license/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms(; experimental=true)
+
+# Rust toolchain for i686 Windows is unusable
+filter!(p -> !Sys.iswindows(p) || arch(p) != "i686", platforms)
+
+# The products that we will ensure are always built
+products = Product[
```
Comment (giordano) [inline C/cargo_license/build_tarballs.jl:35]:
```suggestion
products = [
```

File: C/cargo_license/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    ExecutableProduct("cargo-license", :cargo_license),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
+                compilers=[:c, :rust], preferred_gcc_version=v"7", lock_microarchitecture=false, julia_compat="1.6")
```
Comment (giordano) [inline C/cargo_license/build_tarballs.jl:45]: Are you sure you need GCC 7 and to "unlock" microarchitectures?

File: C/cargo_license/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+script = raw"""
+
+
+cd $WORKSPACE/srcdir/cargo-license
+
+cargo build --release --all
+
+mkdir ${bindir}
+cp target/${rust_target}/release/cargo-license${exeext} ${bindir}/
+
```
Comment (giordano) [inline C/cargo_license/build_tarballs.jl:23]: Why so many (and consecutive) blank lines?

Comment (jeremiahpslewis) [reply C/cargo_license/build_tarballs.jl]: Fixed. :)

**Comment by giordano** [comment]
@jeremiahpslewis please stop starting new jobs, we're having problems on the build machine, I'm doing some cleanup

**Pull request merged**